sound/soc/sof/intel/hda-ipc.c

```diff
@@ -86,11 +86,25 @@ int hda_dsp_ipc_get_reply(struct snd_sof_dev *sdev,
 			  struct snd_sof_ipc_msg *msg)
 {
 	struct sof_ipc_reply reply;
+	struct sof_ipc_cmd_hdr *hdr;
 	int ret = 0;
 	u32 size;
 
-	/* get IPC reply from DSP in the mailbox */
-	sof_mailbox_read(sdev, sdev->host_box.offset, &reply, sizeof(reply));
+
+	hdr = (struct sof_ipc_cmd_hdr *)msg->msg_data;
+	if (hdr->cmd == (SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_CTX_SAVE)) {
+		/*
+		 * memory windows are powered off before sending IPC reply,
+		 * so we can't read the mailbox for CTX_SAVE reply.
+		 */
```

Comment:
Isn't the header in a memory window? How does this work if the memory is no longer powered?

It feels like we actually do need to preserve the use of the msg/msg_ext bits precisely for this case.

Reply:
@plbossart msg->msg_data is dynamic allocated [msg->msg_data = devm_kzalloc(sdev->dev, PAGE_SIZE, GFP_KERNEL);](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/ipc.c#L816). I think it is not in a memory window.